src/iris/vendors/iris_hipchat.py

```diff
@@ -45,16 +43,58 @@ def get_message_payload(self, message):
         }
         return message_dict
 
+    def parse_destination(self, destination):
+        """Determine room_id, token and user to mention
+           We accept 3 formats:
+             - Just a mention (@testuser)
+             - Room_id and Token (12341:a20asdfgjahdASDfaskw)
+             - Room_id, Token and mention (12341:a20asdfgjahdASDfaskw;@testuser)
+        """
+        room_id = self.room_id
+        token = self.token
+        mention = ""
+
+        if destination.startswith("@"):
+            mention = destination
+        elif ";" in destination:
+            dparts = destination.split(";")
+            if len(dparts) == 3:
+                try:
+                    int(dparts[0])
+                    room_id = dparts[0]
+                except ValueError:
+                    pass
```

Comment:
Should this be `pass`? I'm not sure what a non-integer room id would imply. I'd suggest logging a warning that an invalid destination was passed and that the vendor is falling back to the default room.